tiledb/api/src/datatype/strategy.rs

```diff
@@ -210,19 +208,23 @@ mod tests {
             datatype,
         );
         assert_eq!(allowed, r.is_ok(), "try_construct => {:?}", r.err());
-        if let Err(Error::LibTileDB(s)) = r {
-            assert!(
-                s.contains("not a valid Dimension Datatype")
-                    || s.contains("do not support dimension datatype"),
-                "Expected dimension datatype error, received: {}",
-                s
-            );
-        } else {
-            assert!(
-                r.is_ok(),
-                "Found error other than LibTileDB: {}",
-                r.err().unwrap()
-            );
+        match r {
+            Err(Error::LibTileDB(s)) => {
+                assert!(
+                    s.contains("not a valid Dimension Datatype")
+                        || s.contains("do not support dimension datatype"),
+                    "Expected dimension datatype error, received: {}",
+                    s
+                );
+            }
+            Err(Error::Datatype(reason)) => {
```

Comment:
You've confused me with this one. All of the usable datatypes don't return an error so they're not a concern here? We're only checking that the failures are failing because of an invalid type error and not some other random issue. What I *think* you mean is "If the error isn't coming from core, we might miss when they enable a new dimension type" which is reasonable.

I'll update the test to avoid passing all of the string types through with the `DimensionConstraint::StringAscii` which rejects all of those other types.

Reply:
Fixed. Error checking is now 100% untouched so that its core flagging acceptability on the datatype or not.

Reply:
Er, yes, you interpret what I mean correctly. Thank you